src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -30,8 +30,8 @@ protected internal void CallContract(UInt160 contractHash, string method, CallFl
 
             ContractState contract = NativeContract.ContractManagement.GetContract(Snapshot, contractHash);
             if (contract is null) throw new InvalidOperationException($"Called Contract Does Not Exist: {contractHash}");
-            ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod(method);
-            if (md is null) throw new InvalidOperationException($"Method {method} Does Not Exist In Contract {contractHash}");
+            ContractMethodDescriptor md = contract.Manifest.Abi.GetMethod(method, args.Count);
```

Comment:
Maybe we need a method signature, otherwise we can't have two methods with the same number of arguments and different type.

Reply:
That is too complicated.

Reply:
We have the type inside the manifest

Reply:
For example, we have to consider `int` and `ulong` to be the same signature. I think `name` + `pcount` is enough.

Reply:
If we're to have this, I'd rather use full signature also. Of course we can only use types defined in NEP-14, but that should be sufficient.

Reply:
```c#
void Test(UInt160 hash);
void Test(UInt256 hash);
```

How does the `ApplicationEngine` know which method to call, since we only have `ByteString` in the VM?

Reply:
But the compiler will send him to the right signature, if we add signature (as key) in the manifest, it will be the compiler who decide where send you

Reply:
Dynamic call?

Reply:
Agree, dynamic call will break it